whole-tale/dashboard
# Refactor Browse view to match new mockups

# TODO
* Clean up dead code (list view?)
* Testing and more testing

______________________________________________________________________

### Problem
Current styling/layout differs from the mockups of the "Browse" view. 

### Approach
* Restyle cards to match mockups
* Add tab selector for future views (Data + Environments)
* Replace filter dropdown with simple buttons (currently not functional, but should be easy to wire up)
* Add helper text to the top of the view
* Remove `tale-instances` panel from the right side

### How to Test
Prerequisites: at least one Tale created, less than 2 Tales running, clear local storage (for good measure)

1. Checkout and run this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
    * You should be brought to the new browse view, which should nearly match the [f2018 mockups](http://www.crc.nd.edu/~kfurse/wholetale-css-mock-up-sui_f2018/browse.html)
    * You should no longer see the `tale-instances` panel on the right side of the browse view
3. Choose a Tale and click `Run Tale` beneath it
    * You should see a spinner appear over the button, indicating that your Tale is launching
    * You should also see the familiar notification panel, indicating that your Tale is launching
4. Wait for the Tale to finish launching
    * You should see the spinner change to `Stop Tale` after it is done launching
5. Hover over the launched Tale's illustration and click the "View" button that appears
    * You should be taken to the Run view for the chosen Tale
6. Click "Return to Dashboard" at the top-left
    * You should be brought back to the Browse view
7. Click `Run Tale` beneath one or two more Tales (enough to trigger a launch error)
    * You should see the launch error appear in a red error label beneath the button
8. Click `Stop Tale` beneath one of your running Tales
    * A spinner should appear briefly over the button, indicating that the Tale is stopping
    * NOTE: this delay is purely artificial, as the API returns immediately from DELETE /instance calls
9. Wait a few seconds
    * The spinner should disappear after ~6 seconds, leaving you with the original `Run Tale` state
